source/common/router/scoped_config_impl.h

```diff
@@ -52,28 +55,36 @@ class ScopeKey {
   // Caller should guarantee the fragment is not nullptr.
   void addFragment(std::unique_ptr<ScopeKeyFragmentBase>&& fragment) {
     ASSERT(fragment != nullptr, "null fragment not allowed in ScopeKey.");
+    updateHash(*fragment);
     fragments_.emplace_back(std::move(fragment));
   }
 
+  uint64_t hash() const { return hash_; }
   bool operator!=(const ScopeKey& other) const;
-
   bool operator==(const ScopeKey& other) const;
 
 private:
+  // Update the key's hash with the new fragment hash.
+  void updateHash(const ScopeKeyFragmentBase& fragment) {
+    absl::uint128 buffer = absl::MakeUint128(hash_, fragment.hash());
+    hash_ = HashUtil::xxHash64(absl::string_view(reinterpret_cast<char*>(&buffer), 16));
```

Comment:
Nit; can you do this in a more type safe way? Not a fan of unsafe casts like this and assuming a certain size (at least use `sizeof`)